tests/ci_build/lint_cpp.py

```diff
@@ -0,0 +1,173 @@
+        idx = file_path_from_root.find("src/")
+        file_path_from_root = _HELPER.project_name + file_path_from_root[idx + 3 :]
+    else:
+        idx = file_path_from_root.find("include/")
+        if idx != -1:
+            file_path_from_root = file_path_from_root[idx + 8 :]
+        for spath in inc_list:
+            prefix = spath + "/"
+            if file_path_from_root.startswith(prefix):
+                file_path_from_root = re.sub("^" + prefix, "", file_path_from_root)
+                break
+    return re.sub(r"[-./\s]", "_", file_path_from_root).upper() + "_"
+
+
+class Lint:
+    def __init__(self):
+        self.project_name = "xgboost"
+        self.cpp_header_map = {}
+        self.cpp_src_map = {}
+        self.python_map = {}
+
+        self.pylint_cats = set(["error", "warning", "convention", "refactor"])
+        # setup cpp lint
+        cpplint_args = ["--quiet", "--extensions=" + (",".join(CXX_SUFFIX)), "."]
+        _ = cpplint.ParseArguments(cpplint_args)
+        cpplint._SetFilters(
+            ",".join(
+                [
+                    "-build/c++11",
+                    "-build/include,",
+                    "+build/namespaces",
+                    "+build/include_what_you_use",
+                    "+build/include_order",
+                ]
+            )
+        )
+        cpplint._SetCountingStyle("toplevel")
+        cpplint._line_length = 100
+
+    def process_cpp(self, path, suffix):
+        """Process a cpp file."""
+        _cpplint_state.ResetErrorCounts()
+        cpplint.ProcessFile(str(path), _cpplint_state.verbose_level)
+        _cpplint_state.PrintErrorCounts()
+        errors = _cpplint_state.errors_by_category.copy()
+
+        if suffix == "h":
+            self.cpp_header_map[str(path)] = errors
+        else:
+            self.cpp_src_map[str(path)] = errors
+
+    @staticmethod
+    def _print_summary_map(strm, result_map, ftype):
+        """Print summary of certain result map."""
+        if len(result_map) == 0:
+            return 0
+        npass = sum(1 for x in result_map.values() if len(x) == 0)
+        strm.write(f"====={npass}/{len(result_map)} {ftype} files passed check=====\n")
+        for fname, emap in result_map.items():
+            if len(emap) == 0:
+                continue
+            strm.write(
+                f"{fname}: {sum(emap.values())} Errors of {len(emap)} Categories map={str(emap)}\n"
+            )
+        return len(result_map) - npass
+
+    def print_summary(self, strm):
+        """Print summary of lint."""
+        nerr = 0
+        nerr += Lint._print_summary_map(strm, self.cpp_header_map, "cpp-header")
+        nerr += Lint._print_summary_map(strm, self.cpp_src_map, "cpp-source")
+        if nerr == 0:
+            strm.write("All passed!\n")
+        else:
+            strm.write(f"{nerr} files failed lint\n")
+        return nerr
+
+
+_HELPER = Lint()
```

Comment:
Does this need to be a global variable?

Reply:
@ahuber21 , it looks like I have confused you with the lint_cpp.py file. This is just a copy from the xgboost master branch. I don't think we should make any changes for this file in our repo without realy seriouse needs (since it is not a plugin related). In case, we whant to make some changes here, it is more relevant makiing PR to the main XGBoost repo first.